test/e2e/specs/specs-wpcom/wp-editor__schedule.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @group gutenberg
+ */
+
+import {
+	DataHelper,
+	TestAccount,
+	envVariables,
+	GutenbergEditorPage,
+	EditorSettingsSidebarComponent,
+	PublishedPostPage,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( `Editor: Schedule` ), function () {
+	const accountName = envVariables.GUTENBERG_EDGE
+		? 'gutenbergSimpleSiteEdgeUser'
+		: 'simpleSitePersonalPlanUser';
+	const postTitle = `Scheduled Post: ${ DataHelper.getTimestamp() }`;
+	const postContent = DataHelper.getRandomPhrase();
+	let postURL: string;
+	let gutenbergEditorPage: GutenbergEditorPage;
+	let page: Page;
+
+	beforeAll( async function () {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+	} );
+
+	it( 'Go to the new post page', async function () {
+		gutenbergEditorPage = new GutenbergEditorPage( page );
+		await gutenbergEditorPage.visit( 'post' );
+	} );
+
+	it( 'Enter page title', async function () {
+		gutenbergEditorPage = new GutenbergEditorPage( page );
+		await gutenbergEditorPage.enterTitle( postTitle );
+	} );
+
+	it( 'Enter page content', async function () {
+		await gutenbergEditorPage.enterText( postContent );
+	} );
+
+	describe( 'Schedule: future', function () {
+		let editorSettingsSidebarComponent: EditorSettingsSidebarComponent;
+
+		it( 'Open scheduler', async function () {
+			const frame = await gutenbergEditorPage.getEditorFrame();
+			await gutenbergEditorPage.openSettings();
+			editorSettingsSidebarComponent = new EditorSettingsSidebarComponent( frame, page );
+			await editorSettingsSidebarComponent.clickTab( 'Post' );
+			await editorSettingsSidebarComponent.openSchedule();
+		} );
+
+		it( 'Schedule the post for next year', async function () {
+			const currentDate = new Date();
+			await editorSettingsSidebarComponent.schedule( {
+				year: currentDate.getFullYear() + 1,
+				month: ( ( new Date().getMonth() + 1 ) % 12 ) + 1,
```

Comment:
Is the modulo and extra addition necessary here? I may not be following this exactly, but shouldn't we be able to just do `currentDate.getMonth() + 1`? This would give parity with how we approach this below in the past date section!

Reply:
It seems like modulo is required to generate a 0-11 range for the month:

```
> ( ( new Date("2022-01-01").getMonth() + 1 )  ) + 1
13
> ( ( new Date("2022-01-01").getMonth() + 1 ) % 12 ) + 1
1
```

With that said, I re-evaluated this portion in light of your subsequent comment below, and I think I was trying to solve a non-existent problem. I've since removed this part in 2772e4a01442d2997d83025ea17b5806b83a77f6.